src/ActionList/index.ts

```diff
@@ -18,8 +18,8 @@ export type {ActionListLeadingVisualProps, ActionListTrailingVisualProps} from '
  * Collection of list-related components.
  */
 export const ActionList = Object.assign(List, {
-  /** Collects related `Items` in an `ActionList`. */
```

Comment:
To keep backward compatibility, we should not remove Group from ActionList just yet. 

We can tag it as `@deprecated` with a suggestion what to use instead like we do in [PageLayout](https://github.com/primer/react/blob/main/src/PageLayout/PageLayout.tsx#L325-L339).

However, tagging a prop as `@deprecated` might fail the CI in dotcom for all the existing instances, which isn't ideal either (tagging @colebemis to confirm)

Ideally, we want this to be a 3 step process:
1. Keep both API, but recommend Heading instead of Group for new instances
2. Show a warning (not error) for old instances and refactor them overtime
3. Change the warning to error, delete Group from ActionList in next major version

I'm not sure what's the best way to only show a warning, not an error 🤔 (Good candidate for engineering sync)

Reply:
Bonus: because we are only deprecated a subcomponent, we don't need to keep it in `deprecated/v35/group`, we can keep it in the same file and remove it with the next major

Reply:
> However, tagging a prop as @deprecated might fail the CI in dotcom for all the existing instances, which isn't ideal either (tagging @colebemis to confirm)

I don't think deprecated components fail CI. I'm in favor of marking this as `@deprecated` as the first non-breaking step. Then we can follow up and remove it in a major version.